server/activity/activity.controller.js

```diff
@@ -19,16 +20,24 @@ function list({ repository, query, opts }, res) {
     .then(data => res.json({ data }));
 }
 
-function create({ user, repository, body }, res) {
+async function create({ user, repository, body }, res) {
   const outlineConfig = find(getOutlineLevels(repository.schema), { type: body.type });
   const data = {
     ...pick(body, ['uid', 'type', 'parentId', 'position']),
     data: { ...get(outlineConfig, 'defaultMeta', {}), ...body.data },
     repositoryId: repository.id
   };
   const context = { userId: user.id, repository };
-  return Activity.create(data, { context })
-    .then(data => res.json({ data }));
+  const activity = await Activity.create(data, { context });
+
+  if (isTrackedInWorkflow(activity.type)) {
```

Comment:
@underscope upon our discussion yesterday, I've tweaked store getter to ensure tracked activities always have status.
That won't handle the case when workflow statuses/status IDs are changed in the config though, but it seems to me it could be better to handle that in some sort of config migration (in the future).